js/bam/bamIndex.js

```diff
@@ -221,6 +221,102 @@ function optimizeChunks(chunks, lowest) {
     return mergedChunks;
 }
 
+/**
+ * Returns the number of basepairs that a bin spans.
+ */
+function binSpan(bin) {
+    if (bin === 0) {
+        return Math.pow(2, 29);
+    } else if (1 <= bin && bin <= 8) {
+        return Math.pow(2, 26);
+    } else if (9 <= bin && bin <= 72) {
+        return Math.pow(2, 23);
+    } else if (73 <= bin && bin <= 584) {
+        return Math.pow(2, 20);
+    } else if (585 <= bin && bin <= 4680) {
+        return Math.pow(2, 17);
+    } else {
+        return Math.pow(2, 14);
+    }
+}
+
+/**
+ * Returns the basepairs where a bin's children intersect
+ */
+function bpsWhereChildrenIntersect(bin) {
+    const childSpan = binSpan(bin) / Math.pow(2, 3);
+    const numBinsAtLvl = Math.pow(2, 29) / childSpan;
+    const numIntersectionPoints = numBinsAtLvl - 1;
+
+    let ret = [];
+    let offset = childSpan;
+    for (let i = 0; i < numIntersectionPoints; ++i) {
+        ret.push(offset);
+        offset += childSpan;
+    }
+
+    return ret;
+}
+
+/**
+ * Returns true if the given chunk could overlap with the genomic
+ * range [min, max].
+ */
+function chunkCouldOverlapWithRange(chunk, min, max) {
+    // If the chunk begins and ends with a different `coffset`
+    // (see bam spec) then it must intersect multiple bgzf
+    // blocks. In this case, the uncompressed span cannot be
+    // calculated, so pessimistically assume that the chunk
+    // *could* overlap with the query range.
+    if (chunk.minv.block !== chunk.maxv.block) {
+        return true;
+    }
+
+    // If the chunk is in a "leaf" bin of the rtree, we
+    // pessimistically assume that the chunk *could* overlap with
+    // the query range (it's a leaf bin, so it *probably* does
+    // intersect the query range anyway).
+    if (4681 <= chunk.bin && chunk.bin <= 37448) {
```

Comment:
These magic numbers could be renamed to `LEAFBINID_MIN` `LEAFBINID_MAX`

Reply:
OK, thanks for this.  I'm not aware of any good reason for loading the chunks in parallel, if I changed it to load sequentially could we do an optimization closer to the spec recommendation?